modules/platforms/cpp/core-test/src/cluster_node_test.cpp

```diff
@@ -0,0 +1,261 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <boost/test/unit_test.hpp>
+
+#include <ignite/ignition.h>
+#include <ignite/test_utils.h>
+
+using namespace ignite;
+using namespace ignite::common;
+using namespace ignite::common::concurrent;
+using namespace ignite::cluster;
+
+using namespace boost::unit_test;
+
+/*
+ * Test setup fixture.
+ */
+struct ClusterNodeTestSuiteFixture
+{
+    Ignite server;
+    Ignite client;
+
+    /*
+     * Constructor.
+     */
+    ClusterNodeTestSuiteFixture() :
+#ifdef IGNITE_TESTS_32
+        server(ignite_test::StartNode("compute-server0-32.xml", "ClusterTestServer0")),
+        client(ignite_test::StartNode("compute-client-32.xml", "ClusterTestClient"))
+#else
+        server(ignite_test::StartNode("compute-server0.xml", "ClusterTestServer0")),
+        client(ignite_test::StartNode("compute-client.xml", "ClusterTestClient"))
+#endif
+    {
+        // No-op.
+    }
+
+    /*
+     * Destructor.
+     */
+    ~ClusterNodeTestSuiteFixture()
+    {
+        Ignition::Stop(client.GetName(), true);
+        Ignition::StopAll(true);
```

Comment:
StopAll stops the client as well, right? Do we need a separate call for the client above?